main.tf

```diff
@@ -62,9 +62,8 @@ resource "azurerm_storage_account" "this" {
       versioning_enabled            = blob_properties.value.versioning_enabled
 
       dynamic "container_delete_retention_policy" {
-        for_each = blob_properties.value.container_delete_retention_policy == null ? [] : [
-          blob_properties.value.container_delete_retention_policy
-        ]
+        for_each = (
+        blob_properties.value.container_delete_retention_policy != null && try(blob_properties.value.container_delete_retention_policy.enabled, true)) ? [blob_properties.value.container_delete_retention_policy] : []
```

Comment:
blob_properties cannot be null if we are accessing it in this context right? We can remove the null check and the try here.